### ISSUE TITLE
Disable ENA support with EBS builder

### ISSUE THREAD

**Issue by bodgit** [hashicorp/packer#6852]
The problem I'm having is that the latest upstream CentOS 6.x AMI's (mistakenly IMHO) have the ENA support flag set on them, yet don't actually ship with the necessary drivers installed, unlike later CentOS 7.x AMI's. I've found this results in the AMI not working on AWS instance types such as `c5.*`, `m5.*`, etc. (well, it works, but you get an instance with no network interfaces so it's next to useless).

I figured I could build a custom image with Packer and set `ena_support` to `false` to turn it off, however the behaviour seems to be that it doesn't explicitly disable ENA support, it just doesn't enable it in the first place so it inherits the setting from the source AMI and my resulting images still have the flag set.

Is it possible to explicitly disable ENA support when `ena_support` is set to `false`?

**Comment by SwampDragons** [comment]
Interesting. I'd like to reproduce this; which CentOS AMI are you using as a source image?  Can you share a simple template?

**Comment by bodgit** [comment]
This is essentially the template I'm using:
```json
{
  "builders": [
    {
      "access_key": "{{user `aws_access_key`}}",
      "ami_block_device_mappings": [
        {
          "delete_on_termination": true,
          "device_name": "/dev/sda1",
          "volume_size": 10,
          "volume_type": "gp2"
        }
      ],
      "ami_description": "Sourced from {{.SourceAMI}} in {{.BuildRegion}}",
      "ami_name": "Packer CentOS 6 {{isotime | clean_ami_name}}",
      "ami_regions": [
        "eu-west-1",
        "eu-west-2"
      ],
      "ebs_optimized": true,
      "ena_support": false,
      "encrypt_boot": true,
      "iam_instance_profile": "{{user `aws_iam_instance_profile`}}",
      "instance_type": "{{user `aws_instance_type`}}",
      "kms_key_id": "alias/bakery",
      "launch_block_device_mappings": [
        {
          "delete_on_termination": true,
          "device_name": "/dev/sda1",
          "volume_size": 10,
          "volume_type": "gp2"
        }
      ],
      "region": "{{user `aws_region`}}",
      "region_kms_key_ids": {
        "eu-west-1": "alias/bakery",
        "eu-west-2": "alias/bakery"
      },
      "secret_key": "{{user `aws_secret_key`}}",
      "security_group_id": "{{user `aws_security_group_id`}}",
      "source_ami_filter": {
        "filters": {
          "owner-alias": "aws-marketplace",
          "product-code": "6x5jmcajty9edm3f211pqjfn2",
          "virtualization-type": "hvm"
        },
        "most_recent": true,
        "owners": [
          "679593333241"
        ]
      },
      "ssh_keypair_name": "{{user `aws_ssh_keypair_name`}}",
      "ssh_private_key_file": "{{user `ssh_key`}}",
      "ssh_pty": "true",
      "ssh_username": "centos",
      "subnet_id": "{{user `aws_subnet_id`}}",
      "tags": {
        "OS": "CentOS 6",
        "CreatedBy": "Packer",
        "Source": "{{.SourceAMI}}"
      },
      "token": "{{user `aws_session_token`}}",
      "type": "amazon-ebs",
      "vpc_id": "{{user `aws_vpc_id`}}"
    }
  ],
  "min_packer_version": "1.3",
  "provisioners": [
    {
      "inline": [
        "echo 'packer' | sudo -E sed -i '/^HOSTNAME=.*$/d' /etc/sysconfig/network",
        "echo 'packer' | sudo -E rm -rf /etc/udev/rules.d/*persistent*",
        "echo 'packer' | sudo -E rm -rf /etc/ssh/*key*"
      ],
      "type": "shell"
    }
  ],
  "variables": {
    "aws_access_key": "{{env `AWS_ACCESS_KEY_ID`}}",
    "aws_iam_instance_profile": null,
    "aws_instance_type": null,
    "aws_region": "{{env `AWS_DEFAULT_REGION`}}",
    "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
    "aws_security_group_id": null,
    "aws_session_token": "{{env `AWS_SESSION_TOKEN`}}",
    "aws_ssh_keypair_name": null,
    "aws_subnet_id": null,
    "aws_vpc_id": null,
    "ssh_key": null
  }
}
```
The current AMI returned by the `source_ami_filter` is `ami-404f4339` for the `eu-west-1` region.

**Comment by bodgit** [comment]
I've raised PR #6872 which seems to work for me, I've created a CentOS 6.x AMI using the above template and the resulting image now has the ENA flag unset.

The instance type to best test this with is an `r4.*` as that supports both normal and enhanced networking. If you boot the original upstream AMI (or the result of the Packer build without my change) then you get an instance with no functioning networking. If you boot the AMI created by Packer with this change then it now boots with functioning networking.
